pkg/asset/asset.go

```diff
@@ -15,6 +15,9 @@ type Asset interface {
 
 	// Name returns the human-friendly name of the asset.
 	Name() string
+
+	// Load reads the assets within the given directory.
+	Load(dir string) (state *State, found bool, err error)
```

Comment:
can the load provide basic read/stat on files. So each asset doesn't have to remember to prefix `dir`

Reply:
what do you mean? assets are not remembering dir, that's why the dir is passed in.

Reply:
@wking Going to take care of the state of the whole store in the next PR. This one is only going to handle reading individual assets from the disk, the reason is that when run `openshift-install <target>` only the assets in the `<target>` are written to the disk, the state of other assets will be stored in a hidden state file.
That's why we also want individual asset to have a `Load()` function. The implementation of the `Load()` is centralized in the `state.go`, only the master-ignition and network-config are special because they will only look for some specific files under a dir.

Reply:
cc @crawford @staebler this might have to change as `generate` might drop returning `state`?